### ISSUE TITLE
Doesn't seem to work on Firefox

### ISSUE THREAD

**Issue by skorokithakis** [instantpage/instant.page#7]
I know you're aware of this, but can we have an issue to track the problem?

**Comment by dieulot** [comment]
Sure. [I reported the bug to Mozilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1527334). Currently they consider it normal for it not to work. I explained the use case and am waiting for an answer.

**Comment by skorokithakis** [comment]
Thanks for the follow-up! So basically it's not that it doesn't work, they just don't cache the page past what its headers specify. Let's see what happens.

**Comment by TehShrike** [comment]
I added instant.page 1.1.0 to my home page, and the prefetching doesn't seem to be working in Firefox even when cache headers are set: https://joshduff.com/

Works niftily in Chrome, though.

**Comment by TehShrike** [comment]
Never mind – the issue went away when I updated Firefox :-x

**Comment by PikachuEXE** [comment]
@TehShrike What versions are you using? (the one working and the one not working)
Would be good to know which versions of Firefox are compatible with this library (and state it in website)

**Comment by TehShrike** [comment]
It's actually still not working on my Windows machine, on Firefox 65.0.1 :-o

**Comment by dieulot** [comment]
Are you using uBlock Origin? I found out two days ago that it [disables prefetching](https://github.com/gorhill/uBlock/wiki/Dashboard:-Settings#disable-pre-fetching) on Firefox. (I plan on adding a mention on the website.) I [reported this](https://github.com/uBlockOrigin/uBlock-issues/issues/435) to uBlock Origin asking if they could allow same-origin, unfortunately there’s no way.

**Comment by TehShrike** [comment]
@dieulot good call!  I did have uBlock on my Windows machine.  I disabled uBlock and tried again, and it worked as expected.  Thanks for figuring that out.

**Comment by dieulot** [comment]
The [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1527334) has been acknowledged by Mozilla. It’s P3 (low priority) and depends on a refactoring, so I guess we’ll have to wait quite a bit before it’s resolved.

**Comment by Nicholas-Cardot** [comment]
Is there a way to detect if this isn't going to work and disable it when so? Right now, it just pings the server twice, doubling the load on the server. Once for the hover and once for the click.

**Comment by kittenswolf** [comment]
Sounds like a simple useragent check for now

**Comment by dieulot** [comment]
Yeah, I’ll do an option to disable it on Firefox for now.

**Comment by Calinou** [comment]
@dieulot This may be out of scope for instant.page, but wouldn't using `fetch()` instead of prefetching bypass the issue with uBlock Origin on Firefox? `fetch()` has different semantics from `prefetch` though (it's higher-priority), but it should technically be able to do the job.

**Comment by dieulot** [comment]
Good idea @Calinou! I’ll look into it.

**Comment by solarkraft** [comment]
Weird stuff. When uBlock origin is enabled (not for the page, enabled in the browser at all!) prefetching doesn't seem to work. But even when it does (showing up in the network inspector), it's not particularly instant at all - it's like nothing was even preloaded :/
